specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json

```diff
@@ -0,0 +1,637 @@
+        "operationId": "SapMonitors_Delete",
+        "x-ms-long-running-operation": true,
+        "summary": "Deletes a SAP monitor.",
+        "description": "Deletes a SAP monitor with the specified subscription, resource group, and monitor name.",
+        "x-ms-examples": {
+          "Deletes a SAP monitor": {
+            "$ref": "./examples/SapMonitors_Delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SapMonitorNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Deleted SapMonitor"
+          },
+          "202": {
+            "description": "Deleting SapMonitor"
+          },
+          "204": {
+            "description": "No SapMonitor to delete"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "HanaOnAzure"
+        ],
+        "operationId": "SapMonitors_Update",
+        "summary": "Patches the Tags field of a SAP monitor.",
+        "description": "Patches the Tags field of a SAP monitor for the specified subscription, resource group, and monitor name.",
+        "x-ms-examples": {
+          "Update Tags field of a SAP monitor": {
+            "$ref": "./examples/SapMonitors_PatchTags.json"
+          },
+          "Delete Tags field of a SAP monitor": {
+            "$ref": "./examples/SapMonitors_PatchTags_Delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SapMonitorNameParameter"
+          },
+          {
+            "$ref": "#/parameters/TagsParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SapMonitor"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HanaOnAzure/sapMonitors/{sapMonitorName}/providerInstances/{providerInstanceName}": {
+      "get": {
```

Comment:
You should include a collection Get api for the resource type, something like:
"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HanaOnAzure/sapMonitors/{sapMonitorName}/providerInstances"

Reply:
The parent resource (sapMonitor) will show all the providerInstances associated with it, so a list API for providerInstances is unnecessary.

Reply:
@PakDLiu - by ARM contract a collection GET API is mandatory. also, I will check if you are doing that or not, but you should not include the payload of the children in the GET on parent. that has multiple issues